File: project_euler/ruby/euler_2.rb
# Problem #2: Even Fibonacci numbers
def even_fibonacci_numbers
  x = 0
  y = 1
  total_sum = 0

  while x < 4_000_000
    @temp = x
    x = y
    y = @temp + y
    total_sum += x if x.even?
  end

  puts "Sum of all even values is #{total_sum}" # Output: 4613732
end

File: project_euler/js/euler_2.js
// Problem #2: Even Fibonacci numbers
function fibonacci () {
  var x = 0
  var y = 1
  var sum = 0

  do {
    var temp = x
    x = y
    y = temp + y
    if (x % 2 === 0) {
      sum += x
    }
  } while (x < 4000000)
  return sum // Output: 4613732
}
